src/main/scala/mesosphere/marathon/core/launcher/impl/TaskOpFactoryImpl.scala

```diff
@@ -113,10 +109,17 @@ class TaskOpFactoryImpl @Inject() (
     else None
 
     def maybeReserveAndCreateVolumes = if (needToReserve) {
+      val configuredRoles = app.acceptedResourceRoles.getOrElse(config.defaultAcceptedResourceRolesSet)
+      if (configuredRoles != Set(ResourceRole.Unreserved)) {
+        log.warn("Ignoring acceptedResourcesRoles for {}. Only unreserved resources can be reserved", app.id)
```

Comment:
as suggested, the code will now never match if `ResourceRole.Unreserved` is not contained in the configured roles, and log a warning in this case